python/pyspark/sql/types.py

```diff
@@ -71,6 +67,15 @@ def json(self):
                           separators=(',', ':'),
                           sort_keys=True)
 
+    def needSerialization(self):
+        return False
+
+    def serialize(self, obj):
```

Comment:
It might be nice to add docstrings or comments to these methods in order to explain their API contract.  For instance, it looks like `serialize` and `deserialize` aren't actually marshalling to/from bytes or into a format like Pickle but instead are performing type conversions 

I wonder whether we should be a bit more explicit by adding some comments to explain that these methods are essentially converting Python objects to/from "internal" types before pickling them.